hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -628,7 +628,19 @@ func generateOneOfUnionType(ctx context.Context, subschemas []Schema, scanner *S
 		}
 	}
 
-	return astmodel.MakeOneOfType(types...), nil
+	result := astmodel.MakeOneOfType(types...)
+
+	// if the node that contains the oneOf(/anyOf) defines other properties, create an object type with them inside to merge
```

Comment:
You could add a golden file test to ensure this does the right thing?

We have some already in the OneOf folder I think.